src/services/webhook.js

```diff
@@ -298,14 +301,40 @@ class WebhookController {
             .catch(err => {
                 if (err) {
                     if (retryCount < this.retryCount) {
-                        console.error('[WebhookController] Error occurred, trying again:', err);
+                        console.error(`[WebhookController] Error occurred, trying again for ${url}`);
                         this.sendEvents(events, url, retryCount++);
                     } else {
-                        console.error('[WebhookController] Error occurred, max retry count reached:', err);
+                        console.error(`[WebhookController] Error occurred, max retry count reached for ${url}`);
                     }
                 }
             });
     }
+
+    checkOnline() {
+        this.urls.forEach(url => {
+            axios({
+                url: url,
+                method: 'POST',
+                data: { pokemon_id: 0 },
+                headers: {
+                  'Accept': 'application/json',
+                  'Content-Type': 'application/json',
+                  'Cache-Control': 'no-cache',
+                  'User-Agent': 'Nodedradamus',
+                },
+            })
+                .then(() => this.online.add(url))
+                .catch(err => {
+                    if (err) {
+                      this.online.delete(url)
+                      console.warn(`${url} is offline`);
+                    };
+                });
+          });
+        if (webhooks.urls.length > 0 && this.online.size === 0) {
+          console.error('No webhooks are online')
+        }
```

Comment:
Good check. However, this check should be performed only after all the Promises resolve.

On an unrelated note, old polling promises should be cancelled in case of timeout and race conditions.